TestClasses/MetadataTest.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+import struct
+import sys
+import zlib
+from BaseTest import BaseTest
+import subprocess
+
+
+class MetadataTest(BaseTest):
+    def __init__(self):
+        super().__init__()
+        self.fileName = None
+        self.fileType = None
+        self.fileSize = None
+        self.fileLocation = None
+        self.fileAuthor = None
+
+        self.chksmFormat = []
+
+        self.toolCategories = {'xrd' : ['Path:', 'Id:', 'Size:', 'MTime:', 'CTime:', 'ATime:', 'Flags:', 'Mode:', 'Owner:', 'Group:'],
+                               'gfal': ['File:', 'Size:', 'Access:', 'Access:', 'Modify:', 'Change:']}
+
+        self.basecmd = {'stat'      : {'xrd' : ['xrdfs', 'endpoint', 'stat', 'dest'], 
+                                       'gfal': ['gfal-stat', 'endpoint', 'dest', 'args']}, 
+                        'checksum'  : {'xrd' : ['xrdfs', 'endpoint', 'query', 'checksum', 'args', 'dest'], 
+                                       'gfal': ['gfal-sum', 'endpoint', 'dest', 'ADLER32']}                            
+                        }
+
+
+    def subprocess(self, action:str, sourcePath=None, destinBaseNm=None, *args, timeout=None):
+        self.setup(action, sourcePath, destinBaseNm, *args)
+        self.groupedOutput = []
+        self.checksums = []
+        
+        for site, siteVal in self.sitesList.items(): #Iterate over destination sites
+            for prot in self.protocols: #Iterate over protocols
+                
+                tool = self.cmds[prot] #e.g. tool=xrd
+
+                endpoint = prot + siteVal[0] + ':' + str(self.port)
+                ext = '_' + prot.strip(':/') + '_' + tool
+                self.destinPath = siteVal[1] + destinBaseNm
+                
+                fullCmd = self.parse(self.basecmd[self.action][tool], endpoint, self.destinPath+ext, self.args)
+                if timeout is not None:
```

Comment:
if different timeouts are not set, this could be moved to the BaseTest setup instead